app/models/mixins/has_monitoring_manager_mixin.rb

```diff
@@ -0,0 +1,21 @@
+module HasMonitoringManagerMixin
```

Comment:
prepare for troubles with this approach.

* validations and creation time https://github.com/ManageIQ/manageiq-providers-azure/pull/13
* accessing associations through the parent manager https://github.com/ManageIQ/manageiq/pull/12884

Also what happens if you include `HasMonitoringManagerMixin` and `HasNetworkingManagerMixin` ?
It wont chain `ensure_managers`. I was experimenting with `HasParentManager`, which would be included into the parent manager, but it also adds some confusion...
So, in the end, I think this linking of managers via parent_manager is flawed, or at least how we set it up.

Unfortunately there is no better way to do this for now 😢